app/src/main/java/org/wikipedia/imagesearch/ImageSearchFragment.java

```diff
@@ -0,0 +1,198 @@
+package org.wikipedia.imagesearch;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import org.wikipedia.BackPressedHandler;
+import org.wikipedia.R;
+import org.wikipedia.WikipediaApp;
+import org.wikipedia.activity.FragmentUtil;
+import org.wikipedia.history.HistoryEntry;
+import org.wikipedia.page.PageTitle;
+import org.wikipedia.search.SearchInvokeSource;
+import org.wikipedia.settings.Prefs;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+
+/**
+ * Created by amawai on 10/04/18.
+ */
+
+public class ImageSearchFragment extends Fragment implements BackPressedHandler {
```

Comment:
What's the purpose of implement BackPressedHandler for this class?

Reply:
Since the `ImageSearchFragment` is opened from the `MainFragment`, it replaces the screen of the main feed. Back pressing while in the `ImageSearchFragment` without the `BackPressedHandler`, results in being unable to return to the main feed, as it doesn't get destroyed.